cmake/developer_package/compile_flags/functions.cmake

```
@@ -479,3 +479,38 @@ function(ov_target_link_libraries_as_system TARGET_NAME LINK_TYPE)
         set_target_properties(${TARGET_NAME} PROPERTIES NO_SYSTEM_FROM_IMPORTED ON)
     endif()
 endfunction()
+
+#
+# Detects the standard filesystem library and sets the OPENVINO_STD_FS_LIB variable with library name if required.
+#
+function(ov_detect_std_filesystem_lib)
+    if(NOT DEFINED OPENVINO_STD_FS_LIB)
+        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/stdfs/main.cpp
+            "#include <experimental/filesystem>\nint main(int argc, char ** argv) {\n  std::experimental::filesystem::path p(argv[0]);\n  return p.string().length();\n}")
+        try_compile(STD_FS_NO_LIB_NEEDED ${CMAKE_CURRENT_BINARY_DIR}
+                    SOURCES ${CMAKE_CURRENT_BINARY_DIR}/stdfs/main.cpp
+                    COMPILE_DEFINITIONS -std=c++11)
+        try_compile(STD_FS_NEEDS_STDCXXFS ${CMAKE_CURRENT_BINARY_DIR}
+                    SOURCES ${CMAKE_CURRENT_BINARY_DIR}/stdfs/main.cpp
+                    COMPILE_DEFINITIONS -std=c++11
```

```suggestion
                    CXX_STANDARD ${CMAKE_CXX_STANDARD}
```
and we have `CMAKE_CXX_STANDARD` to be `17`

do we need to detect filesystem library at all for C++ 17 ? will not it work out of the box?

---

It is removed.